src/Core/System/SystemConfig/MemoizedSystemConfigLoader.php

```diff
@@ -0,0 +1,52 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\System\SystemConfig;
+
+use Shopware\Core\System\SystemConfig\Event\SystemConfigChangedEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Contracts\Service\ResetInterface;
+
+class MemoizedSystemConfigLoader extends AbstractSystemConfigLoader implements EventSubscriberInterface, ResetInterface
+{
+    private AbstractSystemConfigLoader $decorated;
+
+    /**
+     * @var array[]
+     */
+    private array $configs = [];
+
+    public function __construct(AbstractSystemConfigLoader $decorated)
+    {
+        $this->decorated = $decorated;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            SystemConfigChangedEvent::class => 'reset',
+        ];
+    }
+
+    public function getDecorated(): AbstractSystemConfigLoader
+    {
+        return $this->decorated;
+    }
+
+    public function load(?string $salesChannelId): array
+    {
+        $key = $salesChannelId ?? 'global';
+
+        if (isset($this->configs[$key])) {
+            return $this->configs[$key];
+        }
+
+        $this->configs[$key] = $this->getDecorated()->load($salesChannelId);
+
+        return $this->configs[$key];
+    }
+
+    public function reset(): void
+    {
+        $this->configs = [];
```

Comment:
```suggestion
$key = $event->getSalesChannelId ?? 'global';
unset($this->configs[$key]);
```

to only reset the memoized values for the given sales channel

Reply:
@keulinho I am fine with this. but I just did it this way because it was so in the original implementation but I am happy to improve it.

Reply:
In any case I would need two separate methods then as I also implement `kernel.reset` which does not provide the event but that isn't an issue either.

Reply:
@keulinho I have no made the change as from what I can see the same is the case for the cachedsystemconfigloader.

Reply:
@keulinho OK I re-checked and the CachedConfig is also always invalidated and not doing so can create problems. espeically, if global values change as that also means that inherit values change. I could check if sales channel ID === null and then rreset everything or just reset everything always. I think resetting everything is the safer option.

It is definitely an issue as one test fails otherwise.

Reply:
@keulinho OK. just clearing the one for the sales channel ID can be problematic. as there is the possibility of inherited values. I have now changed it to reset all configs, if the sales channel ID === null and only reset per sales channel if not the global scope.